Source/WebKit/NetworkProcess/NetworkResourceLoader.cpp

```diff
@@ -2181,6 +2182,32 @@ void NetworkResourceLoader::useRedirectionForCurrentNavigation(WebCore::Resource
     m_redirectionForCurrentNavigation = makeUnique<WebCore::ResourceResponse>(WTFMove(response));
 }
 
+void NetworkResourceLoader::sendDidReceiveDataMessage(const FragmentedSharedBuffer& buffer, size_t encodedDataLength)
+{
+    auto delta = bytesTransferredOverNetworkDelta();
+
+    send(Messages::WebResourceLoader::DidReceiveData(IPC::SharedBufferReference(buffer), encodedDataLength, delta));
+
+    if (!delta)
+        return;
+
+    WTF::switchOn(m_parameters.workerIdentifier,
+        [&](std::monostate) { },
+        [&](WebCore::SharedWorkerIdentifier& identifier) {
+            reportNetworkUsageToAllSharedWorkers(identifier, delta);
+        });
+}
+
+void NetworkResourceLoader::reportNetworkUsageToAllSharedWorkers(WebCore::SharedWorkerIdentifier identifier, size_t delta)
+{
+    UNUSED_PARAM(delta);
+    Ref connection = m_connection;
+    if (auto* session = connection->protectedNetworkProcess()->networkSession(sessionID())) {
```

Comment:
I think the safer C++ bot will want this to be `RefPtr session = ...`

Reply:
NetworkSession is not RefCounted but CanMakeCheckedPtr so I will use `CheckedPtr` here.